src/cpp/fastrtps_deprecated/publisher/PublisherImpl.cpp

```diff
@@ -389,7 +490,13 @@ void PublisherImpl::PublisherWriterListener::onWriterChangeReceivedByAll(
         RTPSWriter* /*writer*/,
         CacheChange_t* ch)
 {
-    if (mp_publisherImpl->m_att.qos.m_durability.kind == VOLATILE_DURABILITY_QOS)
+    if (mp_publisherImpl->m_att.topic.topicKind == WITH_KEY &&
+            (NOT_ALIVE_UNREGISTERED == ch->kind ||
+            NOT_ALIVE_DISPOSED_UNREGISTERED == ch->kind))
```

Comment:
Should we also handle `NOT_ALIVE_DISPOSED`?
```suggestion
    if (mp_publisherImpl->m_att.topic.topicKind == WITH_KEY && ALIVE != ch->kind)
```

Reply:
Disposing an instance doesn't release the resources. Only unregistering it.